src/main/java/com/microsoft/azure/util/AzureUtil.java

```diff
@@ -307,15 +307,15 @@ public static boolean isValidTemplateName(String templateName) {
      * @param osType Valid os type
      * @return Valid deployment name to use for a new deployment
      */
-    public static String getDeploymentName(String templateName) {
+    public static String getDeploymentName(String templateName, Date timestamp) {
         if (!isValidTemplateName(templateName)) {
             throw new IllegalArgumentException("Invalid template name");
         }
         
         Format formatter = new SimpleDateFormat(Constants.DEPLOYMENT_NAME_DATE_FORMAT);
         return String.format("%s%s", getShortenedTemplateName(templateName, Constants.USAGE_TYPE_DEPLOYMENT, 
             Constants.DEPLOYMENT_NAME_DATE_FORMAT.length(), 0), 
-                formatter.format(new Date(System.currentTimeMillis())));
+                formatter.format(timestamp));
```

Comment:
Do you think ms precision timestamp is enough? What if two builds start at the same exact time? 
I know the chances are much smaller, but from the user perspective he wouldn't be able to figure out what went wrong. 
Can we use a UUID or the size of that is too large?